main.tf

```diff
@@ -0,0 +1,508 @@
+terraform {
+  required_providers {
+    aws = {
+      source  = "hashicorp/aws"
+      version = "~>5.0"
+    }
+  }
+}
+
+data "aws_region" "current" {}
+
+resource "null_resource" "copy_main_tf" {
+  triggers = {
+    main_tf_content = templatefile("${path.module}/templates/main.tf.tpl", {
+      bucket_name             = var.s3_tf_bucket_name
+      region                  = data.aws_region.current.name
+      dynamodb_table          = var.dynamodb_tf_locks_name
+      security_group_id_input = var.security_group_id != null ? var.security_group_id : ""
+      public_subnet_id_input  = length(var.public_subnet_ids) > 0 ? element(var.public_subnet_ids, 0) : ""
+    })
+  }
+
+  provisioner "local-exec" {
+    command = "echo '${templatefile("${path.module}/templates/main.tf.tpl", {
+      bucket_name             = var.s3_tf_bucket_name
+      region                  = data.aws_region.current.name
+      dynamodb_table          = var.dynamodb_tf_locks_name
+      security_group_id_input = var.security_group_id != null ? var.security_group_id : ""
+      public_subnet_id_input  = length(var.public_subnet_ids) > 0 ? element(var.public_subnet_ids, 0) : ""
+    })}' > ${path.module}/ecr-scripts/main.tf"
```

Comment:
You might want to put this template into a local variable so that we can be a bit more DRY.